test.ps1

```diff
@@ -90,98 +91,65 @@ function test_image ($version, $sdk_tag, $runtime_tag) {
-                $app_build_tag `
-                dotnet publish --configuration Release --output $publish_path $no_restore_flag
-
-            Write-Host "----- Running framework-dependent app with ${runtime_tag} -----"
-            $app_container_name = "runtime-framework-dependent-${app_name}"
-            try {
-                exec docker run -d -t `
-                    --entrypoint dotnet `
-                    --name $app_container_name `
-                    -p ${host_port}:80 `
-                    -v ${app_volume_name}:${publish_path} `
-                    --workdir ${publish_path} `
-                    $runtime_tag `
-                    test.dll
-
-                $ip = Get-Ip $app_container_name $active_os
-                WaitForSuccess "http://${ip}:${host_port}"
-            }
-            finally {
-                exec docker logs $app_container_name
-                exec docker rm -f $app_container_name
-            }
-        }
-        finally {
-            exec docker volume rm $app_volume_name
-        }
+    Write-Host "----- Testing self-contained app with ${sdk_tag} and ${runtime_tag} -----"
 
-        Write-Host "----- Publishing self-contained app with ${sdk_tag} -----"
-        $app_volume_name = "$app_name-self-contained"
-        try {
-            exec docker run --rm `
-                --name "publish-self-contained-$app_name" `
-                -v ${app_volume_name}:${publish_path} `
-                $app_build_tag `
-                dotnet publish --configuration Release --runtime $rid --output $publish_path $no_restore_flag
-
-            if ($active_os -eq "linux" -and $version -eq "2.0") {
-                # Temporary workaround https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/dogfooding.md#option-2-self-contained
-                exec docker run --rm `
-                    -v ${app_volume_name}:${publish_path} `
-                    $runtime_tag `
-                    chmod u+x ${publish_path}/test
-            }
-
-            Write-Host "----- Running self-contained app with ${runtime_tag} -----"
-            $app_container_name = "runtime-self-contained-${app_name}"
-            try {
-                exec docker run -d -t `
-                    --entrypoint $self_contained_entrypoint `
-                    --name $app_container_name `
-                    -p ${host_port}:80 `
-                    -v ${app_volume_name}:${publish_path} `
-                    --workdir ${publish_path} `
-                    $runtime_tag
-
-                $ip = Get-Ip $app_container_name $active_os
-                WaitForSuccess "http://${ip}:${host_port}"
-            }
-            finally {
-                exec docker logs $app_container_name
-                exec docker rm -f $app_container_name
-            }
-        }
-        finally {
-            exec docker volume rm $app_volume_name
-        }
+    $app_tagname = "$app_name-self-contained"
```

Comment:
Consider factoring out the commonality between the self-contained and framework-dependent tests.  The pattern seems identical outside of a couple build args and Dockerfile to build.

Reply:
I'll look into it. I ran into some odd issues with caching in Windows. Possibly a docker bug.